File: packages/db/seed.ts
import { PrismaClient } from './generated/client'

const prisma = new PrismaClient()

async function main() {
  console.log('🌱 Seeding database...')

  // Create test organization
  const testOrg = await prisma.organization.upsert({
    where: { slug: 'test-org' },
    update: {},
    create: {
      name: 'Test Organization',
      slug: 'test-org',
      metadata: {
        industry: 'Technology',
        size: 'Small',
      },
    },
  })

  console.log('✅ Created test organization:', testOrg.name)

  // Create test user
  const testUser = await prisma.user.upsert({
    where: { email: 'admin@test-org.com' },
    update: {},
    create: {
      email: 'admin@test-org.com',
      name: 'Test Admin',
      role: 'OWNER',
      orgId: testOrg.id,
      metadata: {
        source: 'seed',
      },
    },
  })

  console.log('✅ Created test user:', testUser.email)

  // Create test vendors
  const vendors = await Promise.all([
    prisma.vendor.upsert({
      where: { id: 'vendor-1' },
      update: {},
      create: {
        id: 'vendor-1',
        name: 'Acme Supplies',
        email: 'sales@acme-supplies.com',
        phone: '+1-555-0123',
        website: 'https://acme-supplies.com',
        orgId: testOrg.id,
        metadata: {
          specialties: ['Office Supplies', 'Electronics'],
          rating: 4.5,
        },
      },
    }),
    prisma.vendor.upsert({
      where: { id: 'vendor-2' },
      update: {},
      create: {
        id: 'vendor-2',
        name: 'Global Tech Solutions',
        email: 'quotes@globaltech.com',
        phone: '+1-555-0456',
        website: 'https://globaltech.com',
        orgId: testOrg.id,
        metadata: {
          specialties: ['IT Equipment', 'Software'],
          rating: 4.2,
        },
      },
    }),
  ])

  console.log('✅ Created test vendors:', vendors.map(v => v.name).join(', '))

  // Create sample procurement request
  const sampleRequest = await prisma.procurementRequest.create({
    data: {
      title: 'Office Laptop Purchase',
      description: 'Need 10 laptops for new employees',
      items: [
        {
          name: 'Business Laptop',
          quantity: 10,
          specifications: {
            cpu: 'Intel i7 or equivalent',
            ram: '16GB',
            storage: '512GB SSD',
            screen: '14-15 inch',
          },
        },
      ],
      status: 'CREATED',
      priority: 'MEDIUM',
      orgId: testOrg.id,
      createdBy: testUser.id,
      requestedBy: new Date(Date.now() + 14 * 24 * 60 * 60 * 1000), // 2 weeks from now
    },
  })

  console.log('✅ Created sample procurement request:', sampleRequest.title)

  console.log('🎉 Database seeded successfully!')
}

main()
  .catch((e) => {
    console.error('❌ Error seeding database:', e)
    process.exit(1)
  })
  .finally(async () => {
    await prisma.$disconnect()
  })

File: apps/ai-service/seed.py
#!/usr/bin/env python3
"""
Seed script for SupplyGraph MVP
Creates initial data for testing core procurement workflow
"""

import asyncio
import os
from datetime import datetime
from prisma import Prisma
from prisma.enums import Role, RequestStatus

async def main():
    """Seed the database with initial data"""
    
    # Database connection
    database_url = os.getenv("DATABASE_URL", "postgresql://supplygraph:supplygraph123@localhost:5432/supplygraph")
    
    prisma = Prisma()
    await prisma.connect()
    
    try:
        # Create test organization
        org = await prisma.organization.create({
            'name': 'Test Company Inc',
        })
        print(f"✓ Created organization: {org.name}")
        
        # Create test user
        user = await prisma.user.create({
            'email': 'admin@testcompany.com',
            'passwordHash': '$2b$12$LQv3c1yqBWVHxkd0LHAkCOYz6TtxMQJqhN8/LewdBPj6hsxq9w5KS', # password: admin123
            'orgId': org.id,
            'role': Role.ADMIN,
        })
        print(f"✓ Created user: {user.email}")
        
        # Create test vendors
        vendors = []
        vendor_data = [
            {'name': 'Office Supplies Co', 'email': 'sales@officesupplies.com'},
            {'name': 'Tech Equipment Ltd', 'email': 'quotes@techequip.com'},
            {'name': 'Furniture Plus', 'email': 'orders@furnitureplus.com'},
        ]
        
        for vendor_info in vendor_data:
            vendor = await prisma.vendor.create({
                'orgId': org.id,
                'name': vendor_info['name'],
                'email': vendor_info['email'],
                'metadata': {'contact': vendor_info['email'], 'rating': 4.5}
            })
            vendors.append(vendor)
            print(f"✓ Created vendor: {vendor.name}")
        
        # Create sample procurement request
        request = await prisma.procurementrequest.create({
            'orgId': org.id,
            'createdBy': user.id,
            'status': RequestStatus.CREATED,
            'items': {
                'items': [
                    {'name': 'Office Chairs', 'quantity': 10, 'specifications': 'Ergonomic, mesh back'},
                    {'name': 'Standing Desks', 'quantity': 5, 'specifications': 'Adjustable height, 120x60cm'},
                    {'name': 'Monitors', 'quantity': 15, 'specifications': '27 inch, 4K, IPS'}
                ]
            }
        })
        print(f"✓ Created procurement request: {request.id}")
        
        print("\n🎉 Database seeded successfully!")
        print("\n📝 Test Data Created:")
        print(f"   • Organization: {org.name}")
        print(f"   • User: {user.email} (password: admin123)")
        print(f"   • Vendors: {len(vendors)} vendors")
        print(f"   • Procurement Request: {request.id}")
        print("\n🔧 Ready to test core workflow:")
        print("   1. Login at http://localhost:3000")
        print("   2. View procurement request")
        print("   3. Send RFQs to vendors")
        print("   4. Process quotes and approve")
        
    except Exception as e:
        print(f"❌ Error seeding database: {e}")
        raise
    finally:
        await prisma.disconnect()

if __name__ == "__main__":
    asyncio.run(main())